spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/configdata/KubernetesConfigDataLocationResolver.java

```diff
@@ -0,0 +1,130 @@
+	}
+
+	@Override
+	public final int getOrder() {
+		return -1;
+	}
+
+	@Override
+	public final boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
+		return location.hasPrefix(getPrefix())
+				&& (KUBERNETES.isEnforced(context.getBinder()) || KUBERNETES.isDetected(new StandardEnvironment()));
+	}
+
+	@Override
+	public final List<KubernetesConfigDataResource> resolve(ConfigDataLocationResolverContext context,
+			ConfigDataLocation location)
+			throws ConfigDataLocationNotFoundException, ConfigDataResourceNotFoundException {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public final List<KubernetesConfigDataResource> resolveProfileSpecific(
+			ConfigDataLocationResolverContext resolverContext, ConfigDataLocation location, Profiles profiles)
+			throws ConfigDataLocationNotFoundException {
+
+		ConfigDataProperties properties = ConfigDataProperties.of(resolverContext).register();
+
+		HashMap<String, Object> kubernetesConfigData = new HashMap<>();
+		kubernetesConfigData.put("spring.cloud.kubernetes.client.namespace", properties.clientProperties().namespace());
+		String applicationName = resolverContext.getBinder().bind("spring.application.name", String.class).orElse(null);
+		if (applicationName != null) {
+			kubernetesConfigData.put("spring.application.name", applicationName);
+		}
+		PropertySource<Map<String, Object>> propertySource = new MapPropertySource("kubernetesConfigData",
+				kubernetesConfigData);
+		ConfigurableEnvironment environment = new StandardEnvironment();
+		environment.getPropertySources().addLast(propertySource);
+		environment.setActiveProfiles(profiles.getAccepted().toArray(new String[0]));
+		KubernetesNamespaceProvider namespaceProvider = kubernetesNamespaceProvider(environment);
+
+		registerBeans(resolverContext, location, profiles, properties, namespaceProvider);
+
+		KubernetesConfigDataResource resource = new KubernetesConfigDataResource(properties.clientProperties(),
+				properties.configMapProperties(), properties.secretsProperties(), location.isOptional(), profiles,
+				environment);
+		resource.setLog(log);
+
+		return List.of(resource);
+	}
+
+	protected abstract void registerBeans(ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles, ConfigDataProperties properties,
+			KubernetesNamespaceProvider namespaceProvider);
+
+	protected final boolean isRetryEnabledForConfigMap(ConfigMapConfigProperties configMapProperties) {
+		return RETRY_IS_PRESENT && configMapProperties != null && configMapProperties.retry().enabled()
+				&& configMapProperties.failFast();
+	}
+
+	protected final boolean isRetryEnabledForSecrets(SecretsConfigProperties secretsProperties) {
+		return RETRY_IS_PRESENT && secretsProperties != null && secretsProperties.retry().enabled()
+				&& secretsProperties.failFast();
+	}
+
+	private KubernetesNamespaceProvider kubernetesNamespaceProvider(Environment environment) {
```

Comment:
this was protected and overridden in both fabric8 and k8s-native clients, but the overridden version was exactly the same. Most probably an artifact of an initial thought process. I made it private here and dropped the methods in clients.